### PR TITLE
For tracpro

### PR DESCRIPTION
**Pull request by dpoirier**
A few changes that'll help with deploying hosted tracpro:

* Add a state to just install the Postgresql client packages
* Add a state to run a Redis server
* Ensure the virtualenv has the New Relic agent installed if a New Relic key is configured.

### CODE REVIEW

File: project/redis-master.sls [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Install Redis, listening on all interfaces.
+# This does not open the firewall for Redis, so do that elsewhere
```
Comment (mlavin) [inline project/redis-master.sls]: What's the point of listening on all interfaces if you can't access it outside of localhost?

Comment (dpoirier) [inline project/redis-master.sls]: I don't know; I just copied this from hosted-tracpro.  Maybe it just makes it easier to use it from other servers later on if needed?

**Comment by dpoirier** [comment]
Documentation update at https://github.com/caktus/developer-documentation/pull/36

File: project/redis-master.sls (hunk 1 above)
Comment (mlavin) [inline project/redis-master.sls]: Maybe. Feels like just grabbing things out of projects is how this repo got to be in such a mess in the first place. I would say :-1: to opening on all interfaces without configuring the firewall and/or setting a password. See http://antirez.com/news/96

> The Redis security model is: “it’s totally insecure to let untrusted clients access the system, please protect it from the outside world yourself”

Comment (dpoirier) [inline project/redis-master.sls]: Do any current projects need to share a Redis server from multiple hosts?  If not, we could just keep this simple and listen on localhost only, and not complicate it until we have to.

Comment (mlavin) [inline project/redis-master.sls]: All the ones I could think of eventually moved to ElasticCache. Simple sounds good to me.

**Comment by dpoirier** [comment]
Updated Redis to only listen on localhost.

**Comment by mlavin** [comment]
:space_invader:

**Comment by dpoirier** [comment]
Thanks, will wait for a shipit for the corresponding doc updates (https://github.com/caktus/developer-documentation/pull/36) and merge it all at the same time.

**Pull request merged**